enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/mtls/MTLSAuthenticator.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.choreo.connect.enforcer.security.mtls;
+
+import io.opentelemetry.context.Scope;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.ThreadContext;
+import org.wso2.choreo.connect.enforcer.commons.exception.APISecurityException;
+import org.wso2.choreo.connect.enforcer.commons.model.AuthenticationContext;
+import org.wso2.choreo.connect.enforcer.commons.model.RequestContext;
+import org.wso2.choreo.connect.enforcer.constants.APIConstants;
+import org.wso2.choreo.connect.enforcer.constants.APISecurityConstants;
+import org.wso2.choreo.connect.enforcer.security.Authenticator;
+import org.wso2.choreo.connect.enforcer.tracing.TracingConstants;
+import org.wso2.choreo.connect.enforcer.tracing.TracingSpan;
+import org.wso2.choreo.connect.enforcer.tracing.TracingTracer;
+import org.wso2.choreo.connect.enforcer.tracing.Utils;
+import org.wso2.choreo.connect.enforcer.util.FilterUtils;
+
+import java.security.KeyStore;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Map;
+
+/**
+ * Implements the authenticator interface to authenticate request using MTLS.
+ */
+public class MTLSAuthenticator implements Authenticator {
+    private static final Logger log = LogManager.getLogger(MTLSAuthenticator.class);
+    private final boolean isEnableClientValidation = FilterUtils.getClientValidationStatus();
+    private final boolean isClientCertificateEncode = FilterUtils.getCertificateEncodeStatus();
+
+    @Override
+    public boolean canAuthenticate(RequestContext requestContext) {
+        String cert = "";
+
+        if (isEnableClientValidation) {
+            cert = requestContext.getClientCertificate();
+        } else {
+            if (requestContext.getHeaders().containsKey(FilterUtils.getCertificateHeaderName())) {
+                cert = retrieveMTLSHeaderValue(requestContext);
+            }
+        }
+
+        if (cert.isEmpty()) {
+            log.debug("No client certificate is provided");
```

Comment:
Let's add more details to these debug logs.

Reply:
Added the request path

Reply:
Let's not use request path. Might include path params and additional data. Let's log API name, version and `requestContext.getMatchedResourcePath()` instead

Reply:
Changed